packages/salesforcedx-vscode-core/test/jest/decorators/scratch-org-decorator.test.ts

```diff
@@ -0,0 +1,80 @@
+import { ConfigUtil } from '@salesforce/salesforcedx-utils-vscode';
+import { mock } from 'sinon';
+import * as vscode from 'vscode';
+import URI from 'vscode-uri';
+import { monitorOrgConfigChanges, showOrg } from '../../../src/decorators';
+import { nls } from '../../../src/messages';
+
+describe('scratch org decorator', () => {
+  const testUser = 'test@username.com';
+  const browserIcon = `$(browser)`;
+  const openOrgCommand = 'sfdx.force.org.open';
+  const fakePathURI = URI.file('/Users/tester/mockNewFile.test.js');
+  const mockStatusBarItem = {
+    tooltip: '',
+    command: '',
+    show: jest.fn(),
+    text: ''
+  };
+  let onDidCreateEventEmitter: vscode.EventEmitter<vscode.Uri>;
+  let onDidChangeEventEmitter: vscode.EventEmitter<vscode.Uri>;
+  onDidCreateEventEmitter = new vscode.EventEmitter<vscode.Uri>();
+  onDidChangeEventEmitter = new vscode.EventEmitter<vscode.Uri>();
+  const mockWatcher = {
+    // onDidChange: onDidChangeEventEmitter.event,
+    // onDidCreate: onDidCreateEventEmitter.event
+    onDidChange: jest.fn(),
+    onDidCreate: jest.fn()
+  };
+  let mockFileSystemWatcher: jest.SpyInstance;
+  let mockGetDefaultUsername: jest.SpyInstance;
+  let mockCreateStatusBar: jest.SpyInstance;
+
+  beforeEach(() => {
+    mockCreateStatusBar = (vscode.window.createStatusBarItem as any).mockReturnValue(
+      mockStatusBarItem
+    );
+    mockFileSystemWatcher = (vscode.workspace
+      .createFileSystemWatcher as any).mockReturnValue(mockWatcher);
+    mockGetDefaultUsername = jest
+      .spyOn(ConfigUtil, 'getDefaultUsernameOrAlias');
+  });
+
+  describe('show Org', () => {
+    it('should not show the browser icon in the status bar when a default username is not set', async () => {
+      await showOrg();
+      expect(mockCreateStatusBar).toHaveBeenCalled();
+      expect(mockStatusBarItem.text).toEqual('');
+      expect(mockStatusBarItem.tooltip).toEqual(nls.localize('status_bar_open_org_tooltip'));
+      expect(mockStatusBarItem.command).toEqual(openOrgCommand);
+      expect(mockStatusBarItem.show).toHaveBeenCalled();
+      expect(mockGetDefaultUsername).toHaveBeenCalled();
+    });
+    it('should show the browser icon in the status bar when a default username is set', async () => {
+      mockGetDefaultUsername.mockResolvedValue(testUser);
+      await showOrg();
+      expect(mockStatusBarItem.text).toEqual(browserIcon);
+      expect(mockStatusBarItem.show).toHaveBeenCalledTimes(0);
+      expect(mockGetDefaultUsername).toHaveBeenCalled();
+    });
+
+  });
+
+  describe('monitor org changes', () => {
+    it('should show the browser icon when the config file changes', () => {
+      onDidChangeEventEmitter.fire(fakePathURI);
+      // Act
+      monitorOrgConfigChanges();
+      expect(mockFileSystemWatcher).toHaveBeenCalled();
+      expect(mockWatcher.onDidChange).toHaveBeenCalledWith(expect.any(Function));
+      // expect(mockGetDefaultUsername).toHaveBeenCalled();
+    });
```

Comment:
I think the event emitter can be removed from this test file.  By getting a handle on the callback function passed to `watcher.onDidChange()`, we can then just call that callback function, faking like a file change happened.  Let me know if you have questions or would like to huddle to discuss.

<img width="500" alt="2022-11-08_06-46-20" src="https://user-images.githubusercontent.com/46458081/200595995-5472f354-b218-43b9-9b2e-4a27348ef522.png">

Reply:
agreed.  def better from a unit test perspective 👍